cmp/grid/GridModel.ts

```diff
@@ -1043,6 +1045,19 @@ export class GridModel extends HoistModel {
         this.expandLevel = level;
     }
 
+    /**
+     * Get the resolved level labels for the current state of the grid.
+     */
+    get resolvedLevelLabels(): string[] {
+        const {maxDepth, levelLabels} = this,
+            ret = executeIfFunction(levelLabels);
+        if (ret && ret.length < maxDepth + 1) {
+            this.logError('Value produced by `GridModel.levelLabels` has insufficient length.');
+            return null;
+        }
+        return ret ? take(ret, maxDepth + 1) : null;
```

Comment:
A bit curious if there's a reason why we support a levelLabels return that's *longer* than actually required.